# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/kde/kdegames/hotfix-binutils-gcc.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

Hotfix needed for new binutils (bug is in gcc)
(see http://gcc.gnu.org/bugzilla/show_bug.cgi?id=16625)

--- ./libkdegames/kgrid2d.h.orig	2004-08-13 13:52:56.000000000 +0200
+++ ./libkdegames/kgrid2d.h	2004-08-13 13:54:02.000000000 +0200
@@ -291,10 +291,14 @@
         case Right:     return c + Coord( 1,  0);
         case Up:        return c + Coord( 0, -1);
         case Down:      return c + Coord( 0,  1);
+	}
+        switch (n) {
         case LeftUp:    return c + Coord(-1, -1);
         case LeftDown:  return c + Coord(-1,  1);
         case RightUp:   return c + Coord( 1, -1);
         case RightDown: return c + Coord( 1,  1);
+	}
+        switch (n) {
         case Nb_Neighbour: Q_ASSERT(false);
         }
         return c;
